### PR TITLE
mcobbett gherkin unit tests 2

### PR DESCRIPTION
**Pull request by techcobweb**
# Why ?

General refactoring of some bad code smells in the test runners.

- **move some methods from java and gherkin test runner to the abstract runner**
- **move some common methods into the abstract test runner**
- **framework shutdown only called from one place**
- **[Moving the updatestatus and kafka publishing routines into a common abstract test runner](https://github.com/galasa-dev/framework/pull/631/commits/22ee923f483b07993f4d0c2a991fc70791cd3319)**
- **[runTest is now 330 lines long](https://github.com/galasa-dev/framework/pull/631/commits/c586a26afa3fe7fb0ea46b3ec301b7402735965d)**

### CODE REVIEW

**Comment by galasa-team** [comment]
Build successful

**Comment by galasa-team** [comment]
Build successful

**Comment by galasa-team** [comment]
Build successful

**Comment by galasa-team** [comment]
Build successful

**Comment by galasa-team** [comment]
Build successful

File: galasa-parent/dev.galasa.framework/src/main/java/dev/galasa/framework/AbstractTestRunner.java [hunk 1]
```diff
@@ -10,13 +10,31 @@
...
 import dev.galasa.framework.spi.IResultArchiveStore;
+import dev.galasa.framework.spi.IRun;
 import dev.galasa.framework.spi.IShuttableFramework;
+import dev.galasa.framework.spi.ResultArchiveStoreException;
+import dev.galasa.framework.spi.events.TestHeartbeatStoppedEvent;
+import dev.galasa.framework.spi.events.TestRunLifecycleStatusChangedEvent;
+import dev.galasa.framework.spi.teststructure.TestStructure;
+import dev.galasa.framework.spi.utils.DssUtils;
 
 public class AbstractTestRunner {
```
Comment (techcobweb) [inline galasa-parent/dev.galasa.framework/src/main/java/dev/galasa/framework/AbstractTestRunner.java:39]: rename to base test runner

File: galasa-parent/dev.galasa.framework/src/main/java/dev/galasa/framework/AbstractTestRunner.java [hunk 2]
```diff
@@ -78,4 +131,267 @@ protected void recordCPSProperties(
...
+
+    protected void storeRasRunIdInDss(IDynamicStatusStoreService dss, String rasRunId) throws TestRunException {
+        try {
+            this.dss.put("run." + run.getName() + ".rasrunid", rasRunId);
+        } catch (DynamicStatusStoreException e) {
+            throw new TestRunException("Failed to update rasrunid", e);
+        }
+    }
+
+    public static class GalasaStream {
```
Comment (techcobweb) [inline galasa-parent/dev.galasa.framework/src/main/java/dev/galasa/framework/AbstractTestRunner.java:386]: Not supposed to be. there now.

**Comment by galasa-team** [comment]
Build successful

**Review by eamansour — APPROVED** [review]

**Pull request merged**